src/compiler/utilitiesPublic.ts

```diff
@@ -907,7 +907,8 @@ export function getNonAssignedNameOfDeclaration(declaration: Declaration | Expre
             return nameForNamelessJSDocTypedef(declaration as JSDocEnumTag);
         case SyntaxKind.ExportAssignment: {
             const { expression } = declaration as ExportAssignment;
-            return isIdentifier(expression) ? expression : undefined;
+            const skipExpression = skipOuterExpressions(expression);

```

Comment:
Maybe

* `skippedExpression`
* `innerExpression`
* make `expression` a `let` variable and reassign
* use `const expression = skipOuterExpressions((declaration as ExportAssignment).expression);`